### DISCUSSION TITLE
Needed example for Reference Inside server

### DISCUSSION THREAD

**Discussion by savergenio** [open62541pp/open62541pp#286] [General]
Need an example where Reference of server node which also has ref another node from another server.

**Comment by lukasberbuer** [comment]
If you want to reference another server's node, you have to reference it with an `ExpandedNodeId` and a `serverIndex` other than `0` (local).
Now there is another place, where you define the mapping from server indexes to server URNs: The server's `ServerArray` node `"ns=0;i=2254"`.
![image](https://github.com/open62541pp/open62541pp/assets/36054362/60aaef33-03a8-45a7-a0df-bc816e0b7185)

More details are given here: https://github.com/node-opcua/node-opcua/issues/175

The `opcua::Node` class only let you add local references because this is the most common use case. But you can always reach out to the free functions in the `services` namespace which expose the full functionality: https://open62541pp.github.io/open62541pp/group__AddReferences.html
